### ISSUE TITLE
No such file or directory after ClickHouse restart

### ISSUE THREAD

**Issue by liuzhiguo630** [ClickHouse/ClickHouse#37917]
I hava three ClickHouse nodes in kubernetes,  after restart ClickHouse, It seem broken. 

When I run projection's query : `select project, count() as cnt from log.unified_log group by project` I got this exception:
```
std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in file_size: No such file or directory [\"/data/disk-2/store/dce/dcec7b47-1c2e-43fd-9fd8-e52ff404dc2e/20220608_680401_683856_5/p_projects.proj/data.mrk3\"]\nCannot print extra info for Poco::Exception (version 22.5.1.2079 (official build))

std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in file_size: No such file or directory [\"/data/disk-2/store/dce/dcec7b47-1c2e-43fd-9fd8-e52ff404dc2e/20220608_680401_683856_5/p_projects.proj/data.mrk3\"]\nCannot print extra info for Poco::Exception (version 22.5.1.2079 (official build))

std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in file_size: No such file or directory [\"/data/disk-2/store/dce/dcec7b47-1c2e-43fd-9fd8-e52ff404dc2e/20220608_677722_680400_6/p_projects.proj/data.mrk3\"]\nCannot print extra info for Poco::Exception (version 22.5.1.2079 (official build))
```

And I tried rebuild projection, drop and add, but mutation are stucked:

```
SELECT parts_to_do, * FROM system.mutations WHERE is_done = 0;
┌─parts_to_do─┬─database─┬─table───────┬─mutation_id─────────┬─command───────────────────────────┬─────────create_time─┬─block_numbers.partition_id─┬─block_numbers.number─┬─parts_to_do_names──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─parts_to_do─┬─is_done─┬─latest_failed_part──────────┬────latest_fail_time─┬─latest_fail_reason─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│          15 │ log      │ unified_log │ mutation_697648.txt │ DROP PROJECTION p_projects        │ 2022-06-08 15:24:00 │ ['']                       │ [697648]             │ ['20220608_695679_695679_0','20220608_695680_695680_0','20220608_695681_695681_0','20220608_695682_695682_0','20220608_695683_695683_0','20220608_695684_695684_0','20220608_695685_695685_0','20220608_695686_695686_0','20220608_695687_695687_0','20220608_695688_695688_0','20220608_695689_695689_0','20220608_695690_695690_0','20220608_695691_695691_0','20220608_695692_695692_0','20220608_695693_695693_0'] │          15 │       0 │                             │ 1970-01-01 08:00:00 │                                                                                                                                                                                                                                                            │
│          15 │ log      │ unified_log │ mutation_697694.txt │ DROP PROJECTION p_keys            │ 2022-06-08 15:24:27 │ ['']                       │ [697694]             │ ['20220608_695679_695679_0','20220608_695680_695680_0','20220608_695681_695681_0','20220608_695682_695682_0','20220608_695683_695683_0','20220608_695684_695684_0','20220608_695685_695685_0','20220608_695686_695686_0','20220608_695687_695687_0','20220608_695688_695688_0','20220608_695689_695689_0','20220608_695690_695690_0','20220608_695691_695691_0','20220608_695692_695692_0','20220608_695693_695693_0'] │          15 │       0 │                             │ 1970-01-01 08:00:00 │                                                                                                                                                                                                                                                            │
│          15 │ log      │ unified_log │ mutation_697761.txt │ DROP PROJECTION p_logger          │ 2022-06-08 15:25:14 │ ['']                       │ [697761]             │ ['20220608_695679_695679_0','20220608_695680_695680_0','20220608_695681_695681_0','20220608_695682_695682_0','20220608_695683_695683_0','20220608_695684_695684_0','20220608_695685_695685_0','20220608_695686_695686_0','20220608_695687_695687_0','20220608_695688_695688_0','20220608_695689_695689_0','20220608_695690_695690_0','20220608_695691_695691_0','20220608_695692_695692_0','20220608_695693_695693_0'] │          15 │       0 │                             │ 1970-01-01 08:00:00 │                                                                                                                                                                                                                                                            │
│          23 │ log      │ unified_log │ mutation_698002.txt │ DROP PROJECTION p_projects        │ 2022-06-08 15:41:06 │ ['']                       │ [698002]             │ ['20220608_695679_695679_0','20220608_695680_695680_0','20220608_695681_695681_0','20220608_695682_695682_0','20220608_695683_695683_0','20220608_695684_695684_0','20220608_695685_695685_0','20220608_695686_695686_0','20220608_695687_695687_0','20220608_695688_695688_0','20220608_695689_695689_0','20220608_695690_695690_0','20220608_695691_695691_0','20220608_695692_695692_0','20220608_695693_695693_0','20220602_1473_1708_3_697761','20220608_697746_697868_3_697761','20220608_697245_697389_3_697761','20220608_696366_697244_4_697761','20220602_1797_1827_1_697761','20220608_695694_696365_5_697761','20220608_697983_697988_1','20220608_697989_697994_1'] │          23 │       0 │ 20220602_1797_1827_1_697761 │ 2022-06-08 15:57:26 │ std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in file_size: No such file or directory ["/data/disk-3/store/4e9/4e9cbdcf-13c7-4004-87a4-66d9e92a5c33/20220602_1797_1827_1_697761/project.dict.mrk2"]
Cannot print extra info for Poco::Exception (version 22.5.1.2079 (official build)) │
│          23 │ log      │ unified_log │ mutation_698003.txt │ MATERIALIZE PROJECTION p_projects │ 2022-06-08 15:43:16 │ ['']                       │ [698003]             │ ['20220608_695679_695679_0','20220608_695680_695680_0','20220608_695681_695681_0','20220608_695682_695682_0','20220608_695683_695683_0','20220608_695684_695684_0','20220608_695685_695685_0','20220608_695686_695686_0','20220608_695687_695687_0','20220608_695688_695688_0','20220608_695689_695689_0','20220608_695690_695690_0','20220608_695691_695691_0','20220608_695692_695692_0','20220608_695693_695693_0','20220602_1473_1708_3_697761','20220608_697746_697868_3_697761','20220608_697245_697389_3_697761','20220608_696366_697244_4_697761','20220602_1797_1827_1_697761','20220608_695694_696365_5_697761','20220608_697983_697988_1','20220608_697989_697994_1'] │          23 │       0 │ 20220602_1797_1827_1_697761 │ 2022-06-08 15:57:26 │ std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in file_size: No such file or directory ["/data/disk-3/store/4e9/4e9cbdcf-13c7-4004-87a4-66d9e92a5c33/20220602_1797_1827_1_697761/project.dict.mrk2"]
Cannot print extra info for Poco::Exception (version 22.5.1.2079 (official build)) │
└─────────────┴──────────┴─────────────┴─────────────────────┴───────────────────────────────────┴─────────────────────┴────────────────────────────┴──────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────┴─────────┴─────────────────────────────┴─────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
``` 
Now `select * from log.unified_log limit 10 settings skip_unavailable_shards=1`  not working too, 

optimize table not woriking:
```
optimize table log.unified_log

Code: 1001. DB::Exception: Received from localhost:9000. DB::Exception: std::__1::__fs::filesystem::filesystem_error: filesystem error: in file_size: No such file or directory ["/data/disk-3/store/4e9/4e9cbdcf-13c7-4004-87a4-66d9e92a5c33/20220608_695679_695679_0/dt.mrk2"]. (STD_EXCEPTION)
```
Is there some way to fix these data? And why data are broken? I have set `terminationGracePeriodSeconds: 1800` in sts config file, It should be graceful.

**Comment by amosbird** [comment]
What are the definitions of those projections? And what operations did you do? If you can find a way to reproduce, it will be more helpful to find a way to "fix these data".

**Comment by soccerqi** [comment]
I have a similar problem. I have added clause 'settings allow_experimental_projection_optimization=0' in my select query and solved the problem, maybe you can try it out.

**Comment by amosbird** [comment]
Without information on the table schema and projection definitions, it is difficult to determine the root cause of the issue.

**Comment by vilia** [comment]
@amosbird we have faced with the same problem and I'm ready to share necessary for your research details

**Comment by alexey-milovidov** [comment]
No info was provided.

**Comment by vilia** [comment]
few comments from my side:
1. we've faced this issue at least on `23.2.2.20`
2. after creating `test environment` with clickhouse from sources 2023-07-20 we tried to reproduce a bug and failed
3. unfortunately we couldn't make an upgrade because of the performance issues now that's why could check again in production
4. we had this bug with a projection which contains huge amount of entities inside (100mln in one part) during merges

anyway once we upgrade our production to newer version I will share the status & etc
@alexey-milovidov thank you <3
